src/client/hud.cpp

```diff
@@ -862,6 +862,58 @@ void Hud::drawSelectionMesh()
 	}
 }
 
+void Hud::toggleBlockBounds()
+{
+	m_block_bounds_mode++;
+
+	if (m_block_bounds_mode > 2) {
+		m_block_bounds_mode = 0;
+	}
+}
+
+void Hud::drawBlockBounds()
+{
+	if (!m_block_bounds_mode) {
+		return;	
+	}
+
+	video::SMaterial old_material = driver->getMaterial2D();
+	driver->setMaterial(m_selection_material);
+
+	auto pos = player->getStandingNodePos();
+
+	auto blockPos = v3s16(
+		floorf((float) pos.X / 16.0f),
+		floorf((float) pos.Y / 16.0f),
+		floorf((float) pos.Z / 16.0f)
+	);
```

Comment:
```suggestion
	v3s16 blockPos(
		(float)pos.X / MAP_BLOCKSIZE,
		(float)pos.Y / MAP_BLOCKSIZE,
		(float)pos.Z / MAP_BLOCKSIZE
	);
```
1. Use globally valid blocksize macro
2. Replace assignment initialization
3. Floating point truncation is equal to `floorf` and happens automatically for implicit integer conversions

Reply:
Could have sworn there was an edge case involving negative numbers that floorf handled differently than truncation. But maybe that doesn't apply here... Will test.